### PR TITLE
Adding code to remove dependency with AzureKeyVaultServiceEndpointResourceId

### PR DESCRIPTION
**Pull request by jkapil**
Adding code to remove dependency with static value specified in AzureKeyVaultServiceEndpointResourceId of Get-AzureRmEnvironment. The keyvault resource Id is now obtained from server using 401 challenge. The Azure PowerShell common infrastructure currently do not support this in an intuitive fashion, so tweaking its usage by overriding the environment endpoint value before acquiring the token.

### CODE REVIEW

**Comment by azurecla** [comment]
Hi __@jkapil__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <span>
        In order for us to evaluate and accept your PR, we ask that you sign a contribution license agreement. It's all electronic and will take just minutes. I promise there's no faxing. https://cla.azure.com.
    </span>

TTYL, AZPRBOT;

File: src/ResourceManager/KeyVault/Commands.KeyVault/Models/DataServiceCredential.cs [hunk 1]
```diff
@@ -74,14 +84,18 @@ public Task<string> OnAuthentication(string authority, string resource, string s
                 throw new ArgumentException(string.Format(KeyVaultProperties.Resources.UnsupportedAccountType, context.Account.Type));
 
             if (context.Subscription != null && context.Account != null)
-                TenantId = context.Subscription.GetPropertyAsArray(AzureSubscription.Property.Tenants)
+                tenantId = context.Subscription.GetPropertyAsArray(AzureSubscription.Property.Tenants)
```
Comment (markcowl) [inline src/ResourceManager/KeyVault/Commands.KeyVault/Models/DataServiceCredential.cs]: Why not use context.tenant?

**Comment by markcowl** [comment]
@jkapil one minor comment.  I would really like to see a mocked test for the scenario - can you mock this by mocking out the authentication factory?

File: src/ResourceManager/KeyVault/Commands.KeyVault/Models/DataServiceCredential.cs (hunk 1 above)
Comment (jkapil) [inline src/ResourceManager/KeyVault/Commands.KeyVault/Models/DataServiceCredential.cs]: We might, didn't wanted to change logic for tenant ID

**Comment by markcowl** [comment]
@jkapil is this also intended for AzureStack TP2?  If so, you'll need to open a PR against the release-1.2.4 branch

File: src/ResourceManager/KeyVault/Commands.KeyVault/Models/DataServiceCredential.cs (hunk 1 above)
Comment (jkapil) [inline src/ResourceManager/KeyVault/Commands.KeyVault/Models/DataServiceCredential.cs]: Yes this will be integrated into release 1.2.4

**Comment by hovsepm** [comment]
on-demand: http://azuresdkci.cloudapp.net/view/1-AzurePowerShell/job/powershell-demand/973/

**Pull request merged**